lib/database/attendance.ts

```diff
@@ -0,0 +1,125 @@
+import { client } from "../db";
+import { array, TypeOf, Any } from "io-ts";
+import { Attendance, AttendanceSchema, SingleUserAttendance, SingleUserAttendanceSchema, CreateAttendance, CreateAttendanceSchema, createAttendanceArrayType} from "../../models/attendance";
+import { decode } from "io-ts-promise";
+
+const AttendanceArraySchema = array(AttendanceSchema);
+type attendanceArrayType = TypeOf<typeof AttendanceArraySchema>;
+
+const SingleUserAttendanceArraySchema = array(SingleUserAttendanceSchema);
+type singleUserAttendanceArrayType = TypeOf<typeof SingleUserAttendanceArraySchema>;
+
+//get session_ids of events that occur before a given time: GET api/class/[id]/sessions
+type sessionId= {
+    sessionId: string,
+    startStr: string
+}
+const getSessions = async (
+    classId: string,
+    time?: string
+): Promise<sessionId[]> => {
+    let query;
+    if (time){
+        query = {
+            text:
+              "SELECT session_id, start_str "+ 
+              "FROM calendar_information WHERE end_str < $1 "+
+              "AND event_information_id = $2",
+            values: [time, classId],
+          };
+    } else{
+        query = {
+            text:
+              "SELECT session_id, start_str " +
+              "FROM calendar_information "+
+              "WHERE event_information_id = $1",
+            values: [classId],
+        };
+    }
+    const res = await client.query(query);
+    return res.rows;
+};
+
+//get attendance objects for given session id (GET: /api/class/[id]/attendence/[session_id])
```

Comment:
lets get some more descriptive comments here since the sql query is such a giant monster